### ISSUE TITLE
Empty search list when sharee searches a folder that has been renamed by sharer

### ISSUE THREAD

**Issue by nabim777** [owncloud/ocis#7262]
## Describe the bug
When sharee searches the folder after renamed by sharer then the search list of sharee is empty.

## Steps to reproduce

1. As `marie`, create a folder `folder1`
2. As `marie`, share a folder to `einstein`
3. As `einstein`, accept the share
4. As `marie`, rename folder `folder1` to `folder2`
5. As `einstein`, search the folder `folder1` using web UI

## Expected behavior
The `folder1` should be contains in search list
![image](https://github.com/owncloud/ocis/assets/61624650/73bf0afb-0c15-450b-a755-9c1c19ebec42)

## Actual behavior
The search list does not contains `folder1`
![image](https://github.com/owncloud/ocis/assets/61624650/048fd67b-f80c-4eb1-976b-def43fdd1f90)

But 
when `einstein` searches resources by name `folder2`
![image](https://github.com/owncloud/ocis/assets/61624650/fbce17d4-4be7-412e-9100-135b9fa308c8)

**Comment by phil-davis** [comment]
Yes, this is a problem. The back-end search facility needs to (somehow) understand that Einstein's has a folder named `folder1` in his view of his file-system.

And probably the back-end search-engine knows the latest name that Marie has given the folder, and it is indexed somewhere by that name.

And, if Einstein himself renames `folder1` to `folder3` locally for him, then the search-engine needs to find it by the name `folder3` - so there are a few variations of renaming-actions that need to work and be tested.

**Comment by saw-jan** [comment]
CC @ScharfViktor

**Comment by ScharfViktor** [comment]
I see that we have raised this issue here https://github.com/owncloud/ocis/issues/6376 as well.  
oc10 has same behavior: 
![Screenshot 2023-09-12 at 09 54 23](https://github.com/owncloud/ocis/assets/84779829/cfe3bd75-aa81-4199-a751-c62bb2b5ece1)


and based on this comment https://github.com/owncloud/ocis/issues/6376#issuecomment-1611771332, I understand that we want to leave it like it is, so we close issue

**Comment by micbar** [comment]
Correct. Renaming shares does not affect both sides. Everybody can do renames which are not visible to the other share participants.

**Comment by saw-jan** [comment]
Understood, the renaming part is clear.
We need to be clear about the search? bug/feature?
By which name the sharee should search the received file? by the name that appears in the share-with-me page or by the name renamed by sharer of which the sharee don't have idea about? :thinking:

**Comment by phil-davis** [comment]
This issue NOT about the different names of the folder as seen by the sharer and sharee.

It is about what happens when the sharee tries to search for the folder after the sharer has changed the folder name (and the folder name for the sharee is the same as previously)

**Comment by micbar** [comment]
From my pov, the search needs to be consistent with the name in the file list / shared with me list.
